web/views/questionnaire_view.ex

```diff
@@ -29,15 +29,24 @@ defmodule Ask.QuestionnaireView do
     }
   end
 
+  def render("simulation.json", %{simulation: %{respondent_id: respondent_id} = simulation, mode: "mobileweb"}) do
+    render_simulation(simulation)
+    |> Map.put(:index_url, "/mobile/simulation/#{respondent_id}")
+  end
+
   def render("simulation.json", %{simulation: simulation}) do
+    render_simulation(simulation)
+  end
+
+  defp render_simulation(simulation) do
     %{
       respondent_id: simulation.respondent_id,
       simulation_status: simulation.simulation_status,
-      disposition: simulation.disposition,
-      messages_history: simulation.messages_history,
-      submissions: simulation.submissions,
-      current_step: simulation.current_step,
-      questionnaire: render("questionnaire.json", %{questionnaire: simulation.questionnaire})
+      disposition: Map.get(simulation, :disposition),
```

Comment:
Why are these changes needed?

Reply:
I think they were needed because I had removed the `QuestionnaireSimulationStep` struct.

Please take a look at a02f945d. I think it's a better approach.